infrastructure/cosmosdb.tf

```diff
@@ -0,0 +1,31 @@
+resource "azurerm_cosmosdb_account" "cosmos" {
```

Comment:
Great stuff, just a qq regarding the criteria for having resources such as the cosmosdb and keyvault as part of the root module rather than individual modules?

Reply:
I think this is a great thing to talk through during parking lot tomorrow. I'll explain it here for posterity though. 

Modules are generally designed for bits of terraform that are truly reusable.  For example, we may want to create 2 AKS clusters in separate regions but we want the configuration (bar a handful of things like name, region, etc.) to be  the same then we can create a module and feed in the variables.

Within any module, Terraform is smart enough to figure out dependencies between resources and graphs them out smartly to work out what resources can be created at what stage and works in parallel.   If we continuously use modules, we hinder that process and create more work for ourselves by using `output` and `variable` and then manually workout / force the dependencies with `depends_on`.

I love that with modules, the resource as a whole is more concise though, but the minute there are cross dependencies, life starts to get difficult.

For now, are you happy with me closing off this comment and talk more during parking lot tomorrow or would you like to discuss further before you're sure? I genuinely don't mind either way :)